### ISSUE TITLE
SOW: HYRAX - Completion of AllinsonFlex

### ISSUE THREAD

**Issue by jillpe** [notch8/amigos#1]
SoftServ will complete AllinsonFlex (https://github.com/samvera-labs/allinson_flex) work that includes
the development of a UI for administering flexible metadata in the admin dashboard. This work will
include the following steps:
1. Create a DatabaseSchemaLoader that mimics the SimpleSchemaLoader but is called on object
instantiation
2. Add versions to database schemas and track schema versions in work types
3. Add interface to admin area to download and upload schema
4. Port existing AllinsonFlex work versioning UI over to new code
5. Expand DatabaseSchemaLoader to handle M3 (Machine-readable Metadata Modeling) spec
6. Port AllinsonFlex react UI for existing schemas
In response to feedback from Hyku community users, this work will include:
● A UI for editing and creating metadata
● Optional ability to make updates with upload of Yaml file will be retained
● Ability to create and update new work types and metadata profiles
● Ability to create fields or edit/update field names
● Ability to determine which fields are required
● Contexts, a preset of metadata fields related to a specific repository or source, attached to an
admin set, as implemented by Indiana University
● M3 (Machine-readable Metadata Modeling) metadata specification

## Acceptance Criteria
- [ ] A UI for editing and creating metadata is available
- [ ] Optional ability to make updates with upload of Yaml file will be retained
- [ ] Ability to create and update new work types and metadata profiles
- [ ] Ability to create fields or edit/update field names
- [ ] Ability to determine which fields are required
- [ ] Contexts, a preset of metadata fields related to a specific repository or source, attached to an admin set, as implemented by Indiana University
- [ ] M3 (Machine-readable Metadata Modeling) metadata specification

# Testing Instructions
1. Initital
Visit the [metadata profiles tab](https://amigos.hyku-staging.notch8.cloud/metadata_profiles?locale=en). v1.0 matches the basic schema defined by Hyrax/Hyku. 
Manually create a single work from the UI (from profile v1.0)

2. Customizations
Visit the metadata profiles tab again. Download this profile as a start, and add custom properties. 
Upload the custom profile
Create a new work (from profile v2.0)
The form should render the additional properties, as defined in the uploaded profile. 
After saving, properties should render on the show page (if defined to do so)
Note: We are aware there's a bug when removing properties 
  - #54

3. EDITING previously created works
If you Edit a work that was created on a previous profile, its form will no longer reflect the metadata of the profile it was created on. The form and show page should reflect the metadata defined on the latest uploaded profile. If you save the work, it will be saved using the schema of the latest uploaded profile. 
The user will see this noted in the form via the profile badges, like 
<img width="221" alt="image" src="https://github.com/user-attachments/assets/3f126b8c-fa4f-48c5-9e20-0ec45ac2eacf">

## TASKS
- [x] #5
  - [x] https://github.com/scientist-softserv/amigos/issues/15
  - [x] https://github.com/scientist-softserv/amigos/issues/16
  - [x] https://github.com/scientist-softserv/amigos/issues/17
  - [x] https://github.com/scientist-softserv/amigos/issues/18
  - [x] https://github.com/scientist-softserv/amigos/issues/19
  - [x] https://github.com/scientist-softserv/amigos/issues/20
- [x] https://github.com/scientist-softserv/amigos/issues/14 
- [x] #8
- [x] #9
- [x] https://github.com/scientist-softserv/amigos/issues/21

- [x] https://github.com/scientist-softserv/amigos/issues/22
- [x] #23
- [x] #10
- [x] #11
- [x] https://github.com/scientist-softserv/amigos/issues/36

# BUGS
- [ ] #39

# Nice to Haves
- [ ] https://github.com/scientist-softserv/amigos/issues/26
- [x] #24
- [ ] https://github.com/scientist-softserv/amigos/issues/27
- [ ] #12
- [ ] #28
- [x] #29
- [ ] https://github.com/scientist-softserv/amigos/issues/44
- [ ] #45
- [ ] #46
- [ ] https://github.com/scientist-softserv/amigos/issues/33
- [ ] https://github.com/scientist-softserv/amigos/issues/34
- [ ] https://github.com/scientist-softserv/amigos/issues/37

**Comment by ShanaLMoore** [comment]
client review is blocked by #54
